Objects/rangeobject.c

```diff
@@ -572,16 +572,62 @@ range_count(rangeobject *r, PyObject *ob)
 }
 
 static PyObject *
-range_index(rangeobject *r, PyObject *ob)
+range_index(rangeobject *r, PyObject *args, PyObject *kw)
 {
     int contains;
+    static char *keywords[] = {"value", "start", "stop", NULL};
+    PyObject *ob;
+    PyObject *start = NULL, *stop = NULL;
+
+    if (!PyArg_ParseTupleAndKeywords(args, kw, "O|OO:range_index",
+                                    keywords, &ob, &start, &stop)) {
+      return NULL;
+    }
+
+    if (start || stop) {
+        if (start && (!PyLong_CheckExact(start))) {
```

Comment:
It should support any index-like objects. Use `PyNumber_Index()` to convert them to int.